www/js/lib/zimArchive.js

```diff
@@ -98,20 +108,132 @@ function ZIMArchive (storage, path, callbackReady, callbackError) {
                     that.addMetadataToZIMFile('Title')
                 ]).then(function () {
                     console.debug('ZIMArchive metadata loaded:', that);
-                });
+                }).catch(handleError);
             }, 1000); // DEV: If you need any of the above earlier, you can alter this delay
             // We need to get the landing page of any Zimit archive opened
             // Note that the test below catches both zimit and zimit2 types
-            if (/zimit/.test(that.zimType)) {
-                return that.setZimitMetadata().then(function () {
-                    callbackReady(that);
-                });
-            } else {
-                // All listings should be loaded, so we can now call the callback
-                callbackReady(that);
+        // We need to get the landing page of any Zimit archive opened
+        if (/zimit/.test(that.zimType)) {
+            return that.setZimitMetadata()
+                .then(function () {
+                    if (callbackReady) {
+                        callbackReady({ success: true, archive: that });
+                    }
+                })
+                .catch(handleError); // Handle errors during Zimit metadata setting
+        } else {
+            // All listings should be loaded, so we can now call the callback
+            if (callbackReady) {
+                callbackReady({ success: true, archive: that });
             }
-        });
+        }
+    })
+    .catch(handleError); // Handle errors during initial metadata addition
+};
```

Comment:
This looks like an error to me. This brace closes:

`function ZIMArchive (storage, path, callbackReady, callbackError) {`

And therefore orphans all the code underneath it. Please fix.

Reply:
Thank you for the feedback. I'll check it out and make correction